### PR TITLE
Ensure same intersection bounds are always used

### LINKED ISSUE #217: Cannot click the left or right most pixel of an `frame_rect` in puffin_egui
**Describe the bug**
Using Puffin Egui you cannot click the left most or right most pixel of the `frame_rect`

**To Reproduce**
1. Run eframe example under puffin_egui
2. Start profiler
3. Try and click the left most pixel, notice how the hover visualization works but not clicking.

**Expected behavior**
When the hover of an `frame_rect` is visualized I expect to be able to click and select the hovered frame.

**Device:**
 - OS: Windows 11

### PR DESCRIPTION
**Pull request by Hoodad**
### Checklist

* [x] I have read the [Contributor Guide](../CONTRIBUTING.md)
* [x] I have read and agree to the [Code of Conduct](../CODE_OF_CONDUCT.md)
* [x] I have added a description of my changes and why I'd like them included in the section below

### Description of Changes

This ensures the bounds for a `frame_rect` is always the same when checking for intersection in puffin_egui.

Also removed unnecessary call to `response.hovered()`. 

### Related Issues

Fixes #217

### CODE REVIEW

File: puffin_egui/src/lib.rs [hunk 1]
```diff
@@ -689,7 +689,7 @@ impl ProfilerUi {
             };
 
             let frame_rect = Rect::from_min_max(
-                Pos2::new(x, rect.top()),
+                Pos2::new(x - 0.5 * frame_spacing, rect.top() + 0.5 * frame_spacing),
                 Pos2::new(x + frame_width, rect.bottom()),
             );
```
Comment (rib) [inline puffin_egui/src/lib.rs:694]: Could _maybe_ move the `expand2` up to here, like:
```suggestion
            )
            .expand2(vec2(0.5 * frame_spacing, 0.0))
```

but inlining the expansion is probably fine too

**Review by rib — APPROVED** [review]
left a nit comment about maybe moving the previous `expand2` up, but either way this looks good to me 👍

File: puffin_egui/src/lib.rs (hunk 1 above)
Comment (Hoodad) [reply puffin_egui/src/lib.rs:694]: You mean instead of doing the subtraction and addition on line 692?

Comment (rib) [reply puffin_egui/src/lib.rs:694]: right - my impression was that this inlined the equivalent of doing the `.expand2` which was removed where checking if the `mouse_pos` is contained in the frame.

Comment (Hoodad) [reply puffin_egui/src/lib.rs:694]: Yeah I can change that, I also noticed the addition is wrong as it adds to the Y component of the min position. So changing to `.expand2` seems like a good solution 👍

Comment (rib) [reply puffin_egui/src/lib.rs:694]: but on second glance maybe not - this now just expands the `min` position not the `max` position, so `expand2` wouldn't be equivalent.

Comment (Hoodad) [reply puffin_egui/src/lib.rs:694]: The function not expanding max as well is what's wrong that I mentioned in my comment above, I'm fixing that as well.

Comment (rib) [reply puffin_egui/src/lib.rs]: yeah, I realized afterwards that the bars were getting filled out visually if I tried using expand2 up-front and yeah the follow up to reduce the visual rect makes sense

**Review by rib — APPROVED** [review]
looks good to me.

I tested with the self profiling and clicking the left/right pixels seems to work as expected now and the bars are still visually spaced like they were before.

**Pull request merged**